satpy/readers/sgli_l1b.py

```diff
@@ -0,0 +1,266 @@
+https://suzaku.eorc.jaxa.jp/GCOM_C/data/product_std.html
+The live data is available from here:
+https://gportal.jaxa.jp/gpr/search?tab=1
+And the format description is here:
+https://gportal.jaxa.jp/gpr/assets/mng_upload/GCOM-C/SGLI_Level1_Product_Format_Description_en.pdf
+
+"""
+
+import logging
+from datetime import datetime
+
+import dask.array as da
+import h5py
+import numpy as np
+import xarray as xr
+from dask.array.core import normalize_chunks
+from xarray import Dataset, Variable
+from xarray.backends import BackendArray, BackendEntrypoint
+from xarray.core import indexing
+
+# from satpy import CHUNK_SIZE
+from satpy.readers.file_handlers import BaseFileHandler
+
+logger = logging.getLogger(__name__)
+
+resolutions = {"Q": 250,
+               "K": 1000,
+               "L": 1000}
+
+polarization_keys = {0: "0",
+                     -60: "m60",
+                     60: "60"}
+
+
+class HDF5SGLI(BaseFileHandler):
+    """File handler for the SGLI l1b data."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Initialize the filehandler."""
+        super().__init__(filename, filename_info, filetype_info)
+        self.resolution = resolutions[self.filename_info["resolution"]]
+        self.h5file = h5py.File(self.filename, "r")
+
+    @property
+    def start_time(self):
+        """Get the start time."""
+        the_time = self.h5file["Global_attributes"].attrs["Scene_start_time"].item()
+        return datetime.strptime(the_time.decode("ascii"), "%Y%m%d %H:%M:%S.%f")
+
+    @property
+    def end_time(self):
+        """Get the end time."""
+        the_time = self.h5file["Global_attributes"].attrs["Scene_end_time"].item()
+        return datetime.strptime(the_time.decode("ascii"), "%Y%m%d %H:%M:%S.%f")
+
+    def get_dataset(self, key, info):
+        """Get the dataset from the file."""
+        if key["resolution"] != self.resolution:
+            return
+        file_key = info["file_key"]
+        if key["name"].startswith("P"):
+            file_key = file_key.format(polarization=polarization_keys[key["polarization"]])
+        h5dataset = self.h5file[file_key]
+
+        chunks = normalize_chunks(("auto", "auto"), h5dataset.shape, previous_chunks=h5dataset.chunks, dtype=np.float32)
+        dataset = da.from_array(h5dataset, chunks=chunks)
+        attrs = h5dataset.attrs
+
+        dataset = xr.DataArray(dataset, attrs=attrs, dims=["y", "x"])
+        with xr.set_options(keep_attrs=True):
+            if key["name"].startswith(("VN", "SW", "P")):
+                dataset = self.get_visible_dataset(key, dataset)
+            elif key["name"].startswith("TI"):
+                dataset = self.get_ir_dataset(key, dataset)
+            elif key["name"].startswith(("longitude", "latitude")):
+                dataset = self.get_lon_lats(key)
+            elif "angle" in key["name"]:
+                dataset = self.get_angles(key)
+            else:
+                raise NotImplementedError()
```

Comment:
I don't feel very strongly about this, but since code scene is complaining about it, is there a cleaner way to do this? Maybe with a dictionary mapping to functions? That might allow the NotImplementedError to actually be a KeyError which is maybe more clear to the user/log? Either way, maybe an error message should be put in the NotImplementedError?

Reply:
I'd like to avoid having a dictionary as the list of channels and other dataset will be quite large. I'll just extract the method for now and see what code scene says about it. I'm also changing the NotImplementedError into a KeyError with a message.